### ISSUE TITLE
Memory leak and retain cycle in Blake2b

### ISSUE THREAD

**Issue by sammichaels** [Legrandin/pycryptodome#372]
There appears to be a memory leak using Blake2b when using a memoryview object (it may be present elsewhere but I haven't tested).  This is on both macOS and Windows using 3.9.4.

```
>>> from Cryptodome.Hash import BLAKE2b
>>> import gc
>>> import sys
>>> m = memoryview(b'123')
>>> sys.getrefcount(m)
2
>>> h = BLAKE2b.new(data=m, digest_bits=256).digest()
```

At this point the BLAKE2b instance is deallocated and `h` holds bytes.  `m` should have no references from within the code (the two references are python itself).  But releasing the memoryview fails and you can see the reference count has increased, despite the BLAKE2b instance being deallocated.

```
>>> m.release()
Traceback (most recent call last):
  File "<input>", line 1, in <module>
BufferError: memoryview has 1 exported buffer
>>> sys.getrefcount(m)
3
>>> gc.get_referrers(m)[0]['BLAKE2b']
<module 'Cryptodome.Hash.BLAKE2b' from '/redacted/Cryptodome/Hash/BLAKE2b.py'>
```

I discovered excessive memory usage when parsing and hashing files.  A memoryview is a significant performance improvement because I'm passing around many slices of the data in addition to hashing it.  The memoryview in the code originates from:

```
>>> from pathlib import Path
>>> file = Path('/path/to/file')
>>> data = memoryview(file.read_bytes())
```

The memory usage grows with the size of each file.  Each iteration the usage doubles until python  throws a MemoryError.

Using the built in blake2b from hashlib and using it in the exact same manner does not have the retain cycle, allows me to release the memoryview, and deallocates the file bytes as expected.

**Comment by Legrandin** [comment]
Thanks for reporting. When you see this, is the `cffi` module installed?

**Comment by sammichaels** [comment]
No, cffi is not installed.

**Comment by Legrandin** [comment]
OK, thanks. If you have the chance and it's easy enough, could you install `cffi` and repeat the test to see if the memory usage pattern is the same?

Without `cffi`, we fall back onto `ctypes`, so there are some chances the problem is with it.

**Comment by sammichaels** [comment]
Adding cffi seems to have fixed the issue.  It no longer retains a reference to the memoryview and everything is properly deallocated.

**Comment by Legrandin** [comment]
Fixed with 31eba9b0dd492a02ea2c0d2db922ad0ca827c3c7 on master

**Comment by sammichaels** [comment]
You're awesome.  Thank you for all your work with this!

**Comment by Legrandin** [comment]
Fixed in v3.9.5.
